test/test-framework/pkg/apis/cache/v1alpha1/dummy_types.go

```diff
@@ -0,0 +1,114 @@
+// Copyright 2018 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type NoKindSpec struct {
+	// Not included in anything, no kind type
+	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors=true
+	// +operator-sdk:gen-csv:customresourcedefinitions.statusDescriptors=true
+	Size int32 `json:"size"`
+	// Not included in anything, no kind type
+	Boss Hog `json:"hog"`
+}
+type NoKindStatus struct {
+	// Not included in anything, no kind type
+	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors=true
+	// +operator-sdk:gen-csv:customresourcedefinitions.statusDescriptors=true
+	Nodes []string `json:"nodes"`
+}
+
+type DummySpec struct {
+	// Should be in spec
+	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors=true
+	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors.displayName="dummy-pods"
+	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors.x-descriptors="urn:alm:descriptor:com.tectonic.ui:podCount"
+	Size int32 `json:"size"`
+}
+type DummyStatus struct {
+	// Should be in status but not spec, since DummyStatus isn't in DummySpec
+	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors=true
+	// +operator-sdk:gen-csv:customresourcedefinitions.statusDescriptors=true
+	Nodes []string `json:"nodes"`
```

Comment:
It will not pass in the API rules validation.

Reply:
Turned off generators for this file.